### PR TITLE
274 : Trigger Plan evaluation when an event is Added

### PR DESCRIPTION
**Pull request by rehammuzzamil**

### CODE REVIEW

File: src/main/java/org/opensrp/service/EventService.java [hunk 1]
```diff
@@ -150,6 +162,13 @@ public synchronized Event addEvent(Event event) {
 		
 		event.setDateCreated(DateTime.now());
 		allEvents.add(event);
+		String planIdentifier = event.getDetails() != null ? event.getDetails().get("planIdentifier") : null;
+		if (isPlanEvaluationEnabled && planIdentifier != null) {
+			PlanDefinition plan = planRepository.get(planIdentifier);
+			if(plan.getStatus().equals(PlanDefinition.PlanStatus.ACTIVE) && (plan.getEffectivePeriod().getEnd() == null ||
+					plan.getEffectivePeriod().getEnd().isAfter(new DateTime().toLocalDate())))
+			taskGenerator.processPlanEvaluation(plan, null, username);
```
Comment (githengi) [inline src/main/java/org/opensrp/service/EventService.java:170]: Pass in the event to  for plan evaluation

Comment (rehammuzzamil) [reply src/main/java/org/opensrp/service/EventService.java]: Done

File: src/main/java/org/opensrp/service/formSubmission/EventsListener.java [hunk 2]
```diff
@@ -45,6 +45,8 @@
 	private EventsRouter eventsRouter;
 	
 	private ErrorTraceService errorTraceService;
+
+	private String username = "";  //TODO : Pass in a parameter
```
Comment (githengi) [inline src/main/java/org/opensrp/service/formSubmission/EventsListener.java:49]: Remove this todo

File: src/main/java/org/opensrp/service/formSubmission/EventsListener.java [hunk 3]
```diff
@@ -93,7 +95,7 @@ public void processEvent() {
 			
 			for (Event event : events) {
 				try {
-					event = eventService.processOutOfArea(event);
+					event = eventService.processOutOfArea(event, username);
```
Comment (githengi) [inline src/main/java/org/opensrp/service/formSubmission/EventsListener.java:98]: You can pass `event.getProvider` as username here

File: src/main/java/org/opensrp/service/formSubmission/EventsListener.java [hunk 4]
```diff
@@ -140,7 +142,7 @@ private synchronized void addServerVersion() {
 				for (Event event : events) {
 					try {
 						Thread.sleep(1);
-						event = eventService.processOutOfArea(event);
+						event = eventService.processOutOfArea(event,username);
```
Comment (githengi) [inline src/main/java/org/opensrp/service/formSubmission/EventsListener.java:145]: You can pass `event.getProvider` as username here

Comment (rehammuzzamil) [reply src/main/java/org/opensrp/service/formSubmission/EventsListener.java]: Passing event.getProviderId() as username

File: src/main/java/org/opensrp/service/formSubmission/EventsListener.java (hunk 3 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/service/formSubmission/EventsListener.java]: Passing event.getProviderId()

File: src/main/java/org/opensrp/service/formSubmission/EventsListener.java (hunk 2 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/service/formSubmission/EventsListener.java]: Done

File: src/main/java/org/opensrp/service/EventService.java [hunk 5]
```diff
@@ -30,11 +33,20 @@
 	private final EventsRepository allEvents;
 	
 	private ClientService clientService;
+
+	private TaskGenerator taskGenerator;
+
+	private PlanRepository planRepository;
+
+	@Value("#{opensrp['plan.evaluation.enabled']}")
```
Comment (githengi) [inline src/main/java/org/opensrp/service/EventService.java:41]: Add a default of false

Comment (rehammuzzamil) [reply src/main/java/org/opensrp/service/EventService.java:41]: Changed its data type to primitive boolean

**Pull request merged**